api/errors/builder.go

```diff
@@ -0,0 +1,93 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/builder component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeAtLeastOneArg is generated when the parser comes across a
+	// Dockerfile command that doesn't have any args.
+	ErrorCodeAtLeastOneArg = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
Using your example of EXACTLYONEARG vs ATLEASTONEARG - yes I can see taking very different actions based on this information.  One might present the user with one entry field, while the 2nd may present them with a list of entry fields.  But the point is that we shouldn't preclude receivers of these errors from making that choice themselves by merging error codes for different error situations.